oid_validation/test/data/README.md

```diff
@@ -0,0 +1,44 @@
+# Regenerating these certificates
+
+To regenerate the signed certificate in this directory in order to update their expiration dates,
+run the regenerate.sh script in this directory.
+
+The repoauth package has a few SSL-related tests, so rather than repeat all of these steps in that
+dir, the regenerate script copies the newly created cert.crt and valid_ca.crt over to repoauth/test/data
+if that directory is in the expected location relative to this one (as with a git clone). Since the
+repoauth certs are copies of the certs generated here, it's very likely that whatever problem
+requires the certs to be regenerate affects both oid_validation and repoauth, so the certs are
+updated in both places.
+
+
+# Background Information
+
+The pulp_ssl.cnf file is a configuration file created just for these certificates, which pre-
+configures openssl for creating and signing the certificates in this dir.
+
+We create a basic trust chain using the script above. First, a "valid" CA is created, and then
+an "other" CA is created. Then, we create certificates signed by both of the authorities to create
+testable scenarios, such as making sure a certificate is accepted when Pulp requires a cert
+signed by the valid CA, or making sure a certificate is rejected when Pulp requires a cert signed
+by the other CA, etc.
+
+Both the valid and other CA certificates are self-signed, and the `-subj` flag to openssl sets
+the certificate "CN" field to something that makes the two CAs easy to distinguish from one
+another.
+
+Four different types of entitlement certificates are then created, all signed by the "valid" CA.
```

Comment:
Needless quantification, s/Four d/D/ so the readmine doesn't need to be maintained if we add more entitlement scenarios